fix rule hit search #327

jharting opened the pull request:
https://github.com/RedHatInsights/insights-frontend/issues/327
https://github.com/RedHatInsights/insights-frontend/issues/334

I had to wrap listType=card components in a span - otherwise the combination of ng-repeate and transclude was removing the table from the DOM even though there was not reason for that.

Kinlaw commented:
If I search for "an" in the table view, I get all 6 rows. If I then switch to the card view, I see that there are 6 results in the count but I only see 3 cards. The ansible results aren't showing up for some reason.

<img width="1087" alt="screen shot 2017-12-20 at 2 51 49 pm" src="https://user-images.githubusercontent.com/5181010/34225764-8c29fe50-e595-11e7-847e-b0c465556a57.png">

<img width="1087" alt="screen shot 2017-12-20 at 2 51 25 pm" src="https://user-images.githubusercontent.com/5181010/34225772-94d79e18-e595-11e7-918d-54995e49d031.png">

jharting commented:
@Kinlaw should be fixed now. I removed the caching shortcut (which did not work with searching). Also, I switched the getData() function to paginate by default. That should be enough to fix the regression.

If you think we need to have the "fetch all + cache" behavior I just removed I am open to discuss that (in a separate MR/issue)
